src/easyscience/fitting/fitter.py

```diff
@@ -1,193 +1,123 @@
-        else:
-            if (fit_object is not None) or (fit_function is not None):
-                raise AttributeError
-
-        self._engines: List[_C] = minimizers.engines
-        self._current_engine: _C = None
-        self.__engine_obj: _M = None
-        self._is_initialized: bool = False
-        self.create()
-
-        fit_methods = [
-            x
-            for x, y in MinimizerBase.__dict__.items()
-            if (isinstance(y, FunctionType) and not x.startswith('_')) and x != 'fit'
-        ]
-        for method_name in fit_methods:
-            setattr(self, method_name, self.__pass_through_generator(method_name))
-
-        if can_initialize:
-            self.__initialize()
+        self._name_current_minimizer = DEFAULT_MINIMIZER
+        self._minimizer: MinimizerBase  # _minimizer is set in the create method
+        self._update_minimizer(self._name_current_minimizer)
 
-    def _fit_function_wrapper(self, real_x=None, flatten: bool = True) -> Callable:
-        """
-        Simple fit function which injects the real X (independent) values into the
-        optimizer function. This will also flatten the results if needed.
-        :param real_x: Independent x parameters to be injected
-        :param flatten: Should the result be a flat 1D array?
-        :return: Wrapped optimizer function.
-        """
-        fun = self._fit_function
+    def fit_constraints(self) -> list:
+        return self._minimizer.fit_constraints()
 
-        @functools.wraps(fun)
-        def wrapped_fit_function(x, **kwargs):
-            if real_x is not None:
-                x = real_x
-            dependent = fun(x, **kwargs)
-            if flatten:
-                dependent = dependent.flatten()
-            return dependent
+    def add_fit_constraint(self, constraint) -> None:
+        self._minimizer.add_fit_constraint(constraint)
 
-        return wrapped_fit_function
+    def remove_fit_constraint(self, index: int) -> None:
+        self._minimizer.remove_fit_constraint(index)
 
-    def initialize(self, fit_object, fit_function: Callable):
+    def make_model(self, pars=None) -> Callable:
+        return self._minimizer.make_model(pars)
+
+    def evaluate(self, pars=None) -> np.ndarray:
+        return self._minimizer.evaluate(pars)
+
+    def convert_to_pars_obj(self, pars) -> object:
+        return self._minimizer.convert_to_pars_obj(pars)
+
+    def initialize(self, fit_object, fit_function: Callable) -> None:
         """
         Set the model and callable in the calculator interface.
 
         :param fit_object: The EasyScience model object
         :param fit_function: The function to be optimized against.
-        :return: None
         """
         self._fit_object = fit_object
         self._fit_function = fit_function
-        self.__initialize()
+        self._update_minimizer(DEFAULT_MINIMIZER)
 
-    def __initialize(self):
+    def create(self, minimizer_name: str = DEFAULT_MINIMIZER) -> None:
```

Comment:
Yes, you are right, but I did not dare to remove it yet since it is a public method and might be used in other codes.

I will add a code comment that I should be cleaned up.